examples/matrixMultiply.cxx

```diff
@@ -0,0 +1,269 @@
+//   this list of conditions and the disclaimer (as noted below) in the
+//   documentation and/or other materials provided with the distribution.
+//
+// * Neither the name of the LLNS/LLNL nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <iostream>
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/defines.hpp"
+
+void checkSolution(double *C, int in_N);
+
+template <typename T>
+T* allocate(RAJA::Index_type size) {
+  T* ptr;
+#if defined(RAJA_ENABLE_CUDA)
+  cudaMallocManaged((void**)&ptr, sizeof(T)*size,cudaMemAttachGlobal);
+#else
+  ptr = new T[size];
+#endif
+  return ptr;
+}
+
+template <typename T>
+void deallocate(T* &ptr) {
+  if (ptr) {
+#if defined(RAJA_ENABLE_CUDA)
+    cudaFree(ptr);
+#else
+    delete[] ptr;
+#endif
+    ptr = nullptr;
+  }
+}
+
+/*  
+  Example 2: Multiplying Two Matrices
+  
+  ----[Details]--------------------
+  Multiplies two N x N matrices
+  
+  -----[RAJA Concepts]-------------
+  1. Introduces nesting of forall loops (Not currently supported in CUDA)
+  2. Introduces forallN variant of the RAJA loop
+*/
+int main(int RAJA_UNUSED_ARG(argc), char** RAJA_UNUSED_ARG(argv[]))
+{
+
+  std::cout<<"Example 2: Multiplying Two Matrices"<<std::endl;
+
+  const int N  = 1000; 
+  const int NN = N*N;
+
+  double *A = allocate<double>(NN);
+  double *B = allocate<double>(NN);
+  double *C = allocate<double>(NN);
+
+  for(int i=0; i<NN; ++i) {
+    A[i] = 1.0;
+    B[i] = 1.0;
+  }
+
+
+  std::cout<<"Standard C++ Loop"<<std::endl;
+  for(int r=0; r<N; ++r) {
+    for(int c=0; c<N; ++c) {
+
+      int cId = c+r*N; double dot=0.0;
+
+      for(int k=0; k<N; ++k) {
+
+        int aId = k + r*N; 
+        int bId = c + k*N;
+        dot += A[aId]*B[bId];
+      }
+      
+      C[cId] = dot;
+    }        
+  }
+  checkSolution(C,N);
+  std::cout<<"\n"<<std::endl;
+  //======================================
+
+
```

Comment:
@artv3 it may make this example clearer if you kept the outer loop range [0,N) and put a doubly-nested loop inside the lambda. Thoughts?